app/src/components/authentication/DesktopSignIn/DesktopSignIn.js

```
@@ -95,13 +99,13 @@ const DesktopSignIn = () => {
   const handleDoneSignIn = useCallback(
     async (firebaseUser) => {
       setLoading(true);
-      const desktopAuthParams = new URLSearchParams(getDesktopAppAuthParams());
+      const desktopAuthParams = getDesktopAppAuthParams();
       const token = await firebaseUser?.getIdToken();
 
       const isNewUser = params.get("isNewUser");
 
-      const code = desktopAuthParams.get("ot-auth-code");
-      const source = desktopAuthParams.get("source").replace(/ /g, "_");
+      const code = desktopAuthParams?.get("ot-auth-code");
```

If `desktopAuthParams` is null here then code will be `undefined`, and we seem to be proceeding with it. Which I don't think we should, can we throw an error here? And desktop auth?